### PR TITLE
Crew Monitor Icons

### PR DESCRIPTION
**Pull request by Zergspower**
Adds icons to the Crew Monitor to spread out the colors as well as add specific icons for death/critical.

## About The Pull Request

Changes the crew monitor tgui to be a little more friendly on the eyes, making it easier to tell the differences relating to individual crew health.

## How This Contributes To The Skyrat Roleplay Experience

Makes it easier for those with color challenges to spot the few-shades-of-red-difference between somewhat hurt, critical and dead. I also gave a Question mark to Binary trackers rather than it just saying N/A. Binary sensors will state Alive/Dead still alongside a subsequent skull or heart.

## Changelog
image_2022-04-13_185755405
image_2022-04-13_185740537

- Adds icons to states of life
- Adds icons to N/A tracking on Binary sensors
- Changes the math to better suit SkyRat's player health pool rather than TG's 
- Changes the harmed state to be yellow-green rather than a slightly darker shade of green from healthy
- Changes the death state to be a darker read rather than a slightly darker shade of red from Crit

:cl:
add: icons to all states of living on the Crew Monitor
expansion: Critical and death now have a visual icon difference
del: old attribute math and gave it something more dynamic
/:cl:

### CODE REVIEW

File: tgui/packages/tgui/interfaces/CrewConsole.js [hunk 1]
```diff
@@ -164,7 +192,7 @@ const CrewTableEntry = (props, context) => {
         )}
       </Table.Cell>
       <Table.Cell>
-        {area !== undefined ? area : 'N/A'}
+        {area !== undefined ? area : <Icon name="question" color="#ffffff" size={1} /> }
```
Comment (Zonespace27) [inline tgui/packages/tgui/interfaces/CrewConsole.js]: skyrat comment

File: tgui/packages/tgui/interfaces/CrewConsole.js [hunk 2]
```diff
@@ -1,16 +1,26 @@
...
-  '#2ecc71',
+  '#c4cf2d', // SKYRAT EDIT - Original'#2ecc71' - moved to make it visually different,
   '#e67e22',
   '#ed5100',
   '#e74c3c',
-  '#ed2814',
+  '#801308', // SKYRAT EDIT - Original'#ed2814' - darker to help distinguish better,
+];
+
+const HEALTH_ICON_BY_LEVEL = [// SKYRAT EDIT ADDITION  - Icon status list
```
Comment (Zonespace27) [inline tgui/packages/tgui/interfaces/CrewConsole.js]: comment at start/end

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

File: tgui/packages/tgui/interfaces/CrewConsole.js (hunk 1 above)
Comment (Zergspower) [reply tgui/packages/tgui/interfaces/CrewConsole.js]: ah yeah that was a last minute one i overlooked

File: tgui/packages/tgui/interfaces/CrewConsole.js [hunk 3]
```diff
@@ -90,11 +102,11 @@ const CrewTable = (props, context) => {
         <Table.Cell bold collapsing textAlign="center">
           Vitals
         </Table.Cell>
-        <Table.Cell bold>
+        <Table.Cell bold collapsing textAlign="center">
```
Comment (Zonespace27) [inline tgui/packages/tgui/interfaces/CrewConsole.js]: sr edit comment

File: tgui/packages/tgui/interfaces/CrewConsole.js [hunk 4]
```diff
@@ -90,11 +102,11 @@ const CrewTable = (props, context) => {
...
           Position
         </Table.Cell>
-        {!!data.link_allowed && (
-          <Table.Cell bold collapsing>
+        {(
+          <Table.Cell bold collapsing textAlign="center">
```
Comment (Zonespace27) [inline tgui/packages/tgui/interfaces/CrewConsole.js]: ditto

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

**Comment by theselfish** [comment]
Any reason you're not submitting this upstream?

**Comment by Zergspower** [comment]
> Any reason you're not submitting this upstream?

it's modifying our existing non-modular crewmonitor

**Comment by Zergspower** [comment]
Last integration error CI Suite / Integration Tests (pull_request)  is for a map which this doesn't touch

**Comment by theselfish** [comment]
> > Any reason you're not submitting this upstream?
> 
> it's modifying our existing non-modular crewmonitor

If it's non modular, then you should do it upstream.

**Comment by Zergspower** [comment]
> > > Any reason you're not submitting this upstream?
> > 
> > 
> > it's modifying our existing non-modular crewmonitor
> 
> If it's non modular, then you should do it upstream.

I misspoke, it wont work downstream without multiple other file changes relating to flags for robots.

Ref https://github.com/Skyrat-SS13/Skyrat-tg/pull/12147

**Comment by E231kei500bandai** [comment]
Hey I was porting this over to TG and noticed there is an error with this PR.
You removed `!!data.link_allowed && ` but this was actually an important piece of code. 
This code made sure that the Tracking column only appeared for AIs because only AIs have the ability to track. Now that you removed it, the Tracking column will appear for non-AIs and present an empty column,

**Comment by Zergspower** [comment]
> Hey I was porting this over to TG and noticed there is an error with this PR. You removed `!!data.link_allowed && ` but this was actually an important piece of code. This code made sure that the Tracking column only appeared for AIs because only AIs have the ability to track. Now that you removed it, the Tracking column will appear for non-AIs and present an empty column,

odd, havn't noticed the extra column - will look closer

**Pull request merged**